src/shogun/distance/Distance.h

```diff
@@ -120,6 +120,29 @@ class CDistance : public CSGObject
 			return distance(idx_a, idx_b);
 		}
 
+		/**  
+		 * Precomputation related to features of right hand side
+		 * WARNING : Make sure to reset computations using reset_precompute()
```

Comment:
This needs a bit more documentation.
Put the warnings you have in Euclidean distance here as well. Also mention this method in the class description on top.